src/components/app/details/cdDetails/deploymentHistoryDiff/DeploymentHistoryDetailedView.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react'
+import { Progressing, showError } from '../../../../common'
+import DeploymentHistoryHeader from './DeploymentHistoryHeader'
+import { getDeploymentHistoryDetail, prepareHistoryData } from '../service'
+import { useParams } from 'react-router'
+import {
+    DeploymentTemplateOptions,
+    CompareViewDeploymentType,
+    DeploymentHistoryDetail,
+    DeploymentHistoryParamsType,
+} from '../cd.type'
+import DeploymentHistorySidebar from './DeploymentHistorySidebar'
+import DeploymentHistoryDiffView from './DeploymentHistoryDiffView'
+
+export default function DeploymentHistoryDetailedView({
+    showTemplate,
+    setShowTemplate,
+    deploymentHistoryList,
+    setDeploymentHistoryList,
+}: CompareViewDeploymentType) {
+    const { appId, pipelineId, historyComponent, baseConfigurationId, historyComponentName } =
+        useParams<DeploymentHistoryParamsType>()
+    const [selectedDeploymentTemplate, setSelectedDeploymentTemplate] = useState<DeploymentTemplateOptions>()
+    const [currentConfiguration, setCurrentConfiguration] = useState<DeploymentHistoryDetail>()
+    const [baseTemplateConfiguration, setBaseTemplateConfiguration] = useState<DeploymentHistoryDetail>()
+    const [previousConfigAvailable, setPreviousConfigAvailable] = useState<boolean>(true)
+    const [loader, setLoader] = useState<boolean>(true)
+
+    useEffect(() => {
+        if (selectedDeploymentTemplate) {
+            setLoader(true)
+            if (selectedDeploymentTemplate.value === 'NA') {
+                setLoader(false)
+            } else {
+                try {
+                    getDeploymentHistoryDetail(
+                        appId,
+                        pipelineId,
+                        selectedDeploymentTemplate.value,
+                        historyComponent,
+                        historyComponentName,
+                    ).then((response) => {
+                        setCurrentConfiguration(prepareHistoryData(response.result, historyComponent))
+                        setLoader(false)
+                    })
+                } catch (err) {
+                    showError(err)
+                    setLoader(false)
+                }
+            }
+        }
+    }, [selectedDeploymentTemplate])
+
+    useEffect(() => {
+        try {
+            setLoader(true)
+            setSelectedDeploymentTemplate(null)
+            setCurrentConfiguration(null)
+            getDeploymentHistoryDetail(
+                appId,
+                pipelineId,
+                baseConfigurationId,
+                historyComponent,
+                historyComponentName,
+            ).then((response) => {
+                setBaseTemplateConfiguration(prepareHistoryData(response.result, historyComponent))
+            })
```

Comment:
Also, set set loader to false when API call is successful?

Reply:
We are setting loader false in next step